ui/src/app/views/dataset/component.tsx

```diff
@@ -58,15 +92,102 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
       const {namespace, id} = this.props.workspace;
       const conceptSets = await Promise.resolve(conceptSetsApi()
         .getConceptSetsInWorkspace(namespace, id));
-      this.setState({conceptSetList: conceptSets.items});
+      this.setState({conceptSetList: conceptSets.items, loadingConceptSets: false});
     } catch (error) {
       console.log(error);
     }
   }
 
+  convertResource(r: ConceptSet | Cohort, rType: ResourceType): RecentResource {
+    const {workspace} = this.props;
+    this.setState({rType: rType});
+    console.log(rType);
+    return convertToResource(r, workspace.namespace, workspace.id,
+      workspace.accessLevel as unknown as WorkspaceAccessLevel, rType);
+  }
+
+  openConfirmDelete(r: ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    this.setState({confirmDeleting: true, resource: rc});
+  }
+
+  closeConfirmDelete(): void {
+    this.setState({confirmDeleting: false});
+    if (this.state.rType === ResourceType.CONCEPT_SET) {
+      const deleted = this.state.resource.conceptSet;
+      const updatedList = fp.filter(conceptSet => conceptSet.id !== deleted.id,
+        this.state.conceptSetList);
+      this.setState({conceptSetList: updatedList});
+    }
+    this.setState({resource: undefined, rType: undefined});
+  }
+
+  edit(r: Cohort | ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    console.log(rType);
+    this.setState({editing: true, resource: rc});
+  }
+
+  receiveDelete() {
+    const {resource, rType} = this.state;
+    let call;
+    const id = rType === ResourceType.COHORT ? resource.cohort.id : resource.conceptSet.id;
+    const argList = [resource.workspaceNamespace,
+      resource.workspaceFirecloudName, id];
+    if (rType === ResourceType.CONCEPT_SET) {
+      // these also throw errors in the logs
+      // functions correctly, and I think is so pretty, but
+      // I don't like all the errors in the logs, so torn.
```

Comment:
I think errors in the logs is fine for now, if they are real errors. Eventually we will probably want to handle more cleanly. 

Do they throw errors even on success?

Reply:
Not sure what you mean by "on success", but from what I can tell about the UI tests in circle is that this fails strict compilation, so will probably have to pull it out. Going to consult with Brett first though.

Reply:
fixed